chainer/variable.py

```diff
@@ -241,6 +273,12 @@ def zerograd(self):
         warnings.warn(
             'Variable.zerograd is deprecated. Use Variable.cleargard instead.',
             DeprecationWarning)
+
+        if self.data is None:
+            dtype = getattr(self.initializer, 'dtype', None)
+            self._grad_initializer = initializers.Zero(dtype)
+            return
```

Comment:
As `zerograd` is deprecated, I think it is better to remove `zerograd` in major version up, rather than fixing it.